### PR TITLE
Remove brush shape

### LINKED ISSUE #2942: Remove deprecated brush_shape
## 🧰 Task

Following on from #2426 by @ppwadhwa, are we now ready to remove the deprecated brush shape and only allow for the square brush? It will simplify the codebase. @jni if you're good for this to go now then @ppwadhwa you could pick this up when you have time - should be pretty easy once green light from @jni. We'd just be left with the circle brush as the only brush type after this.

### PR DESCRIPTION
**Pull request by ppwadhwa**
Following #2426, this will remove the label square brush shape completely.  

Closes #2942

### CODE REVIEW

**Comment by ppwadhwa** [comment]
I am not sure, but I don't think the failing tests are related.

File: napari/layers/labels/labels.py [hunk 1]
```diff
@@ -549,35 +549,12 @@ def show_selected_label(self, filter):
     @property
     def brush_shape(self):
         """str: Paintbrush shape"""
```
Comment (sofroniewn) [inline napari/layers/labels/labels.py]:
```suggestion
```
I think we can get rid of the whole public `layer.brush_shape` attribute and `layer._brush_shape` too

File: napari/layers/labels/labels.py [hunk 2]
```diff
@@ -549,35 +549,12 @@ def show_selected_label(self, filter):
...
-            ),
-            category=FutureWarning,
-            stacklevel=2,
-        )
-
         return str(self._brush_shape)
 
     @brush_shape.setter
     def brush_shape(self, brush_shape):
         """Set current brush shape."""
```
Comment (sofroniewn) [inline napari/layers/labels/labels.py]:
```suggestion
```

File: napari/layers/labels/labels.py [hunk 3]
```diff
@@ -1066,30 +1043,7 @@ def paint(self, coord, new_label, refresh=True):
...
-                    1,
-                )
-                for c, s, brush_size in zip(
-                    coord, self.data.shape, brush_size_dims
-                )
-            )
-            slice_coord = tuple(map(np.ravel, np.mgrid[slice_coord]))
-            slice_coord = indices_in_shape(slice_coord, shape)
-        elif str(self._brush_shape) == "circle":
+        if str(self._brush_shape) == "circle":
```
Comment (sofroniewn) [inline napari/layers/labels/labels.py]:
```suggestion
```
Can cut too

File: napari/layers/labels/labels.py [hunk 4]
```diff
@@ -1124,6 +1078,13 @@ def paint(self, coord, new_label, refresh=True):
 
             slice_coord = tuple(slice_coord)
 
+        else:
+            warnings.warn(
+                trans._(
+                    "The only valid brush_shape is a circle.",
+                )
+            )
```
Comment (sofroniewn) [inline napari/layers/labels/labels.py]:
```suggestion
```
Can cut

File: napari/layers/labels/_labels_constants.py [hunk 5]
```diff
@@ -57,19 +57,15 @@ class LabelBrushShape(StringEnum):
     LabelBrushShape: Labelling brush shape.
 
     CIRCLE (default) uses circle paintbrush (case insensitive).
-
-    SQUARE uses square paintbrush (case insensitive).
     """
 
     CIRCLE = auto()
```
Comment (sofroniewn) [inline napari/layers/labels/_labels_constants.py]: can cut too

**Review by sofroniewn — COMMENTED** [review]
Off to a good start @ppwadhwa, but I think we can go all the way and drop `layer.brush_shape` and `layer._brush_shape` entirely now - we'll forever just be hardcoded at a circle. Sorry i didn't make that clear. Thanks!!

**Comment by ppwadhwa** [comment]
> I think we can go all the way and drop layer.brush_shape and layer._brush_shape entirely now - we'll forever just be hardcoded at a circle. Sorry i didn't make that clear.

I should have asked!  I had a brief thought about it, then didn't follow up.  Thanks for clarifying.  :)

**Comment by ppwadhwa** [comment]
@sofroniewn I think I got everything this time.  Ready for review again.  :)

File: napari/layers/labels/_tests/test_labels_mouse_bindings.py [hunk 6]
```diff
@@ -27,17 +27,13 @@ def Event():
     )
 
 
-@pytest.mark.parametrize(
-    "brush_shape, expected_sum", [("circle", 244), ("square", 274)]
-)
-def test_paint(Event, brush_shape, expected_sum):
-    """Test painting labels with circle/square brush."""
+@pytest.mark.parametrize("expected_sum", [(244)])
```
Comment (sofroniewn) [inline napari/layers/labels/_tests/test_labels_mouse_bindings.py]: Can you drop the parametrize, i think can just hard code `[(244)]` as answer now

File: napari/layers/labels/_tests/test_labels_mouse_bindings.py [hunk 7]
```diff
@@ -69,16 +65,13 @@ def test_paint(Event, brush_shape, expected_sum):
     assert np.sum(layer.data == 3) == expected_sum
 
 
-@pytest.mark.parametrize(
-    "brush_shape, expected_sum", [("circle", 244), ("square", 274)]
-)
+@pytest.mark.parametrize("brush_shape, expected_sum", [("circle", 244)])
```
Comment (sofroniewn) [inline napari/layers/labels/_tests/test_labels_mouse_bindings.py]: can you also drop parametrize here too, can just hard code `[(244)]`

File: napari/layers/labels/_tests/test_labels_mouse_bindings.py [hunk 8]
```diff
@@ -110,21 +103,17 @@ def test_paint_scale(Event, brush_shape, expected_sum):
...
-    "brush_shape, expected_sum", [("circle", 156), ("square", 126)]
-)
+@pytest.mark.parametrize("brush_shape, expected_sum", [("circle", 156)])
 def test_erase(Event, brush_shape, expected_sum):
     """Test erasing labels with different brush shapes."""
     data = np.ones((20, 20), dtype=np.int32)
     layer = Labels(data)
-    with pytest.warns(FutureWarning):
-        layer.brush_shape = brush_shape
+    layer.brush_shape = brush_shape
```
Comment (sofroniewn) [inline napari/layers/labels/_tests/test_labels_mouse_bindings.py]: these `layer.brush_shape` can go too, multiple in this file

File: napari/layers/labels/_tests/test_labels.py [hunk 9]
```diff
@@ -699,34 +694,30 @@ def test_paint_2d(brush_shape, expected_sum):
...
     layer.brush_size = 12
-    with pytest.warns(FutureWarning):
-        layer.brush_shape = brush_shape
     layer.mode = 'paint'
     layer.paint((1, 1, 512, 512), 3)
     assert isinstance(layer.data, xr.DataArray)
     assert layer.data.sum() == expected_sum
 
 
 @pytest.mark.parametrize(
```
Comment (sofroniewn) [inline napari/layers/labels/_tests/test_labels.py]: can drop parametrize

File: napari/layers/labels/_tests/test_labels.py [hunk 10]
```diff
@@ -619,62 +619,57 @@ def test_show_selected_label():
...
     assert np.unique(layer.data[:3, :3]) == 1
 
 
-@pytest.mark.parametrize(
-    "brush_shape, expected_sum",
-    [("circle", [41, 137, 137, 41, 349]), ("square", [36, 144, 169, 36, 400])],
-)
-def test_paint_2d(brush_shape, expected_sum):
-    """Test painting labels with circle/square brush."""
+@pytest.mark.parametrize("expected_sum", [([41, 137, 137, 41, 349])])
```
Comment (sofroniewn) [inline napari/layers/labels/_tests/test_labels.py]: can drop parameterize

**Review by sofroniewn — COMMENTED** [review]
Looking good, still left a couple things that can be dropped, but getting close!!

**Comment by ppwadhwa** [comment]
OK, ready again.   :)

**Review by sofroniewn — APPROVED** [review]
I think this is it! Good stuff @ppwadhwa! I'm approving now, will merge this tomorrow. The property was already deprecated, but we'll have to remember to put the removal notice in our release notes

**Comment by sofroniewn** [comment]
I'm actually going to merge this now as it will allow some simplifications too #3050. I think there's been pretty extensive discussion around this PR on previous issues so I hope no one minds the merge now. Feel free to flag though if there's something here someone disagrees with. Thanks!!

**Pull request merged**